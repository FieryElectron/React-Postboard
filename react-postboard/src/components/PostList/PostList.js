import SinglePost from '../SinglePost/SinglePost';


const PostList = () => {
    return (
        <div>
            <SinglePost/>
            <SinglePost/>
            <SinglePost/>
        </div>
    )
}

export default PostList;