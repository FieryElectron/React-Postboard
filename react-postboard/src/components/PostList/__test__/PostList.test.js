import ReactDOM from "react-dom/client";
import { render, fireEvent  } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom/extend-expect';

import PostList from '../PostList';


let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

it("renders postList without crashing", ()=>{
    act(() => {
        ReactDOM.createRoot(container).render(<PostList/>);
    });
});


it("matchs snapshot 1", ()=>{
  const tree = renderer.create(<PostList/>).toJSON();
  expect(tree).toMatchSnapshot();
});