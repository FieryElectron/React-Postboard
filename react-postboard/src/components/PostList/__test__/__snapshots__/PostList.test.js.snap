// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs snapshot 1 1`] = `
<div>
  <div
    className="singlePostContainer"
    data-testid="singlePostContainer"
  >
    <div
      className="ownerName"
      data-testid="ownerName"
    >
      ownerName
    </div>
    <div
      className="postContent"
      data-testid="postContent"
    >
      Content
    </div>
    <hr />
    <div
      className="timeStampContainer"
      data-testid="timeStampContainer"
    >
      <span
        className="timeStamp"
        data-testid="timeStamp"
      >
        timestamp
      </span>
      <button
        className="button"
        data-testid="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E62E2E",
          }
        }
      >
        Delete
      </button>
    </div>
  </div>
  <div
    className="singlePostContainer"
    data-testid="singlePostContainer"
  >
    <div
      className="ownerName"
      data-testid="ownerName"
    >
      ownerName
    </div>
    <div
      className="postContent"
      data-testid="postContent"
    >
      Content
    </div>
    <hr />
    <div
      className="timeStampContainer"
      data-testid="timeStampContainer"
    >
      <span
        className="timeStamp"
        data-testid="timeStamp"
      >
        timestamp
      </span>
      <button
        className="button"
        data-testid="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E62E2E",
          }
        }
      >
        Delete
      </button>
    </div>
  </div>
  <div
    className="singlePostContainer"
    data-testid="singlePostContainer"
  >
    <div
      className="ownerName"
      data-testid="ownerName"
    >
      ownerName
    </div>
    <div
      className="postContent"
      data-testid="postContent"
    >
      Content
    </div>
    <hr />
    <div
      className="timeStampContainer"
      data-testid="timeStampContainer"
    >
      <span
        className="timeStamp"
        data-testid="timeStamp"
      >
        timestamp
      </span>
      <button
        className="button"
        data-testid="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E62E2E",
          }
        }
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
